design-system/react/src/components/InputAmount/InputAmount.tsx

```diff
@@ -73,57 +82,97 @@ export const InputAmount: InputAmountComponent = ({
-        decimalScale={units}
-        {...inputProps}
-      />
-      {initialBalance && (
-        <Input.ElementRight css={styles.elementRight}>
-          <Box css={styles.balanceActions}>
-            {!hiddenMaxButton && (
-              <Flex align="center" justify="end">
+      <Text color="intentsBase9" css={styles.heading}>
+        Amount
+      </Text>
+      <Flex align="center" css={styles.secondRow}>
+        <Input.Number
+          autoComplete="off"
+          inputMode="decimal"
+          name={name}
+          aria-label={name}
+          placeholder="0.00"
+          allowedDecimalSeparators={['.', ',']}
+          allowNegative={false}
+          thousandSeparator={false}
+          value={assetAmount}
+          onChange={(e) => {
+            handleAmountChange(e.target.value);
+          }}
+          decimalScale={units}
+          {...inputProps}
+        />
+
+        {initialBalance && (
+          <Input.ElementRight css={styles.elementRight}>
+            <Box css={styles.balanceActions}>
+              {!hiddenMaxButton && (
+                <Flex align="center">
+                  <Button
+                    aria-label="Max"
+                    variant="solid"
+                    intent="primary"
+                    onPress={handleSetBalance}
+                    css={styles.maxButton}
+                  >
+                    MAX
+                  </Button>
+                </Flex>
+              )}
+              {asset && onClickAsset && (
                 <Button
-                  aria-label="Max"
-                  size="sm"
-                  variant="ghost"
-                  onPress={handleSetBalance}
-                  css={styles.maxButton}
+                  size="xs"
+                  aria-label="Coin Selector"
+                  variant="outlined"
+                  intent="base"
+                  css={styles.assetButton}
+                  onPress={onClickAsset}
                 >
-                  Max
+                  {tokenImage(asset.name, asset.imageUrl)}
+                  <Text fontSize="xs" color="intentsBase12">
+                    {asset.name}
+                  </Text>
+                  <Icon
+                    icon="CaretDown"
+                    size="10"
+                    color="intentsBase12"
+                    css={{ color: '$intentsBase12 !important' }}
+                  />
                 </Button>
-              </Flex>
-            )}
-            {!hiddenBalance && (
-              <Flex
-                as="div"
-                css={styles.balance}
-                aria-label={`Balance: ${formattedBalance}`}
-              >
-                <Box as="span">Balance: </Box>
-                <Tooltip
-                  content={formatAmount(balance, formatOpts)}
-                  sideOffset={-5}
-                >
-                  <Box as="span">{formattedBalance}</Box>
-                </Tooltip>
-              </Flex>
-            )}
+              )}
```

Comment:
to increase this component flexibility, let's include other use case: when has asset, but doesn't support multiple assets

1- if there's `asset` and no `onClickAsset`, should show the button still, but in readOnly mode . also should hide "caretDown" icon
2- if there's `asset` and `onClickAsset`, show in the way it is now: button active to click + "caretDown" icon